src/sentry/utils/eventuser.py

```diff
@@ -3,43 +3,152 @@
+        Fetch the EventUser with a Snuba query that exists within a list of projects
+        and valid `keyword_filters`. The `keyword_filter` keys are in `KEYWORD_MAP`.
+        """
+        oldest_project = min(projects, key=lambda item: item.date_added)
+
+        where_conditions = [
+            Condition(Column("project_id"), Op.IN, [p.id for p in projects]),
+            Condition(Column("timestamp"), Op.LT, datetime.now()),
+            Condition(Column("timestamp"), Op.GTE, oldest_project.date_added),
+        ]
+
+        for keyword, value in keyword_filters.items():
+            snuba_column = SNUBA_KEYWORD_MAP.get_key(keyword)
+            if isinstance(snuba_column, tuple):
+                for column in snuba_column:
+                    where_conditions.append(Condition(Column(column), Op.EQ, value))
+            else:
+                where_conditions.append(Condition(Column(snuba_column), Op.EQ, value))
+
+        query = Query(
+            match=Entity(EntityKey.Events.value),
+            select=[
+                Column("project_id"),
+                Column("group_id"),
+                Column("ip_address_v6"),
+                Column("ip_address_v4"),
+                Column("event_id"),
+                Column("user_id"),
+                Column("user"),
+                Column("user_name"),
+                Column("user_email"),
+            ],
+            where=where_conditions,
+            limit=Limit(1),
+            orderby=[OrderBy(Column("timestamp"), Direction.DESC)],
+        )
+
+        request = Request(
+            dataset=Dataset.Events.value,
+            app_id=REFERRER,
+            query=query,
+            tenant_ids={"referrer": REFERRER, "organization_id": projects[0].organization.id},
+        )
+        data_results = raw_snql_query(request, referrer=REFERRER)["data"]
+        results = [EventUser.from_snuba(result) for result in data_results]
+
+        return results
+
+    @staticmethod
+    def from_snuba(result: Mapping[str, Any]) -> EventUser:
+        """
+        Converts the object from the Snuba query into an EventUser instance
+        """
+        return EventUser(
+            id=None,
+            project_id=result.get("project_id"),
+            email=result.get("user_email"),
+            username=result.get("user_name"),
+            name=result.get("user_name"),
+            ip_address=result.get("ip_address_4") or result.get("ip_address_6"),
+            user_ident=result.get("user_id"),
+        )
+
+    @property
+    def tag_value(self):
+        """
+        Return the identifier used with tags to link this user.
+        """
+        for key, value in self.iter_attributes():
+            if value:
+                return f"{KEYWORD_MAP[key]}:{value}"
+
+    def iter_attributes(self):
+        """
+        Iterate over key/value pairs for this EventUser in priority order.
+        """
+        for key in KEYWORD_MAP.keys():
+            yield key, getattr(self, key)
```

Comment:
I just checked where `iter_attributes` is used, and it seems like it's really only used internally here. Just wanted to be sure that changing `ident` to `user_ident`, doesn't cause problems, since it also changes the values of `key` emitted here.